packages/databases-collections/src/components/rename-collection-modal/rename-collection-modal.tsx

```diff
@@ -98,21 +94,16 @@ function RenameCollectionModal({
     [setNewName, clearError]
   );
   const onFormSubmit = () => {
-    if (modalState === 'confirmation-screen') {
-      setModalState('input-form');
-      renameCollection(newName);
-    } else {
-      setModalState('confirmation-screen');
-    }
+    submitModal(newName);
   };
 
   useTrackOnChange(
     (track: TrackFunction) => {
-      if (isVisible) {
+      if (modalState !== 'hidden') {
```

Comment:
Not sure if this is an expected change in logic here: we're going to be reporting the same screen for both `input` and `confirmation` form types.

Might be that it would be a bit easier to reason about this code if we still calculae `isVisible` somewhere in the render function scope and continue to use this variable for the logic in other parts of the component where exact modal state is not that important maybe?

Reply:
oh right, thanks for noticing that!